### PR TITLE
Avoid exception in case of closing a file watched by notifier, like t…

### PR DESCRIPTION
**Pull request by nishidy**
…rapping SIGINT to close the file without exception.

To continue after watching file is stopped without exception.

```
trap :INT do
    notifier.close
    file.close
end
 
notifier.run

# continue after watching file is stopped
...

```

### CODE REVIEW

**Comment by nishidy** [comment]
#47

**Comment by nishidy** [comment]
#47

File: lib/rb-inotify/notifier.rb [hunk 1]
```diff
@@ -274,6 +275,7 @@ def read_events
         tries += 1
         retry
       end
+      return [] if data.nil?
```
Comment (nex3) [inline lib/rb-inotify/notifier.rb]: We should probably mention that this is possible in the method comment.

File: lib/rb-inotify/notifier.rb [hunk 2]
```diff
@@ -292,7 +294,11 @@ def read_events
     # Same as IO#readpartial, or as close as we need.
     def readpartial(size)
       # Use Ruby's readpartial if possible, to avoid blocking other threads.
-      return to_io.readpartial(size) if self.class.supports_ruby_io?
+      begin
+        return to_io.readpartial(size) if self.class.supports_ruby_io?
+      rescue Errno::EBADF
```
Comment (nex3) [inline lib/rb-inotify/notifier.rb]: Add a comment explaining when this can happen and why returning `nil` is correct.

**Comment by nex3** [comment]
This looks pretty good! Just a little more documentation and I'll merge it in.

**Comment by nishidy** [comment]
Sorry for my late response. Updated as you suggested. Check, please.

**Pull request merged**